Fix calendar widget resize when adding or removing other widgets

b12f opened the pull request:
The fullcalendar dependency only reacts to the `document.resize` event. However in our case the widget can also resize
when neighboring widgets are added or removed. This fix adds a `ResizeObserver` listener to account for that. It is
supported in all evergreen browsers.

It also moves all `@types` to the devdependencies.

Issue: https://community.openproject.com/projects/openproject/work_packages/34429/activity

Commit:
Fix calendar widget resize when adding or removing other widgets

The fullcalendar dependency only reacts to the `document.resize` event. However in our case the widget can also resize
when neighboring widgets are added or removed. This fix adds a `ResizeObserver` listener to account for that. It is
supported in all evergreen browsers.

openprojectci commented:
<!--
  0 Errors
  1 Warning: <a href="https://github.com/op...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/9159779316f0746a075c5f335eed5009481344d9/frontend/src/app/modules/calendar/wp-calendar/wp-calendar.component.ts#L38">frontend/src/app/modules/calendar/wp-calendar/wp-calendar.component.ts#L38</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

b12f commented:
> 	1 Warning
> warning 	[frontend/src/app/modules/calendar/wp-calendar/wp-calendar.component.ts#L38](https://github.com/opf/openproject/blob/610a68659a930aeb1ca1657498f330c5f21a50ed/frontend/src/app/modules/calendar/wp-calendar/wp-calendar.component.ts#L38) - Please use `ChangeDetectionStrategy.OnPush` for this component
> 
> Generated by no_entry_sign [Danger](https://danger.systems/)

ulferts reviewed (changes requested):
Two things:

* I checked the behaviour without the fix (using release/11.0) and at least on my machine, it worked just fine, meaning that the calendar is resized as expected (MacOS + FF/Chrome). Now it might be that there is some combination of os/browser which does not work but this should be double checked.
* While the type declaration for `ResizeObserver` is added via `resize-observer-browser` that declaration is not imported where it is required. Because of that, npm reports an error and because of that, the specs fail.

I did not know `ResizeObserver` before and think it could be rather nifty on other places of the application as well. Thanks, for bringing it to my knowledge.

Commit:
Merge branch 'dev' into fix/fix-calendar-resize

Commit:
Explicitly add resize observer typings

The typings file for ResizeObserver was not being autoloaded. This commit adds the typings explicitly, making sure tsc
picks up on them.

Commit:
Move typings include to a different file
